src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -106,6 +106,51 @@ public class RuntimeConfigLoaderJsonDeserializerTests
             }
         }
 
+        /// <summary>
+        /// Tests the parsing of the runtime configuration with environment variables based on replaceEnvVar parameter.
+        /// </summary>
+        /// <param name="replaceEnvVar">A boolean indicating whether to replace environment variables in the configuration.</param>
+        [DataTestMethod]
+        [DataRow(false, DisplayName = "Do not replace environment variables.")]
+        [DataRow(true, DisplayName = "Replace environment variables.")]
+        public void TestConfigParsingWithEnvVarReplacement(bool replaceEnvVar)
+        {
+            SetEnvironmentVariablesFromText(_environmentFileContent);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(
```

Comment:
Don't we have any existing tests in ConfigurationTests that would check environment variables are correctly replaced? Why do we need to add a new test?

Reply:
the existing ones doesn't check the DataSource section.

Reply:
i created new tests since there are different cases that needs to be tested for the data-source section. So, separating tests made more sense to debug if any failures occur because of future changes.